### PR TITLE
HTTP2 now supports request bodies

### PR DESCRIPTION
**Pull request by tamird**
Also updates generated code for gRPC.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,37 +8,39 @@ Currently covered:
...
+|------------------|-------:|--------------:|------------:|------------:|---------------:|
+| GRPC_1K-4        | 100000 |   74513 ns/op |  27.48 MB/s |  16055 B/op	|  106 allocs/op |
+| GRPC_64K-4       |  10000 |  858855 ns/op | 152.61 MB/s | 714106 B/op	|  346 allocs/op |
+| GobRPC_1K-4      | 200000 |   40553 ns/op |  50.50 MB/s |   4745 B/op	|   33 allocs/op |
+| GobRPC_64K-4     |  20000 |  421501 ns/op | 310.96 MB/s | 263899 B/op	|   97 allocs/op |
+| ProtoRPC_1K-4    | 200000 |   33851 ns/op |  60.50 MB/s |   2632 B/op	|   27 allocs/op |
+| ProtoRPC_64K-4   |  20000 |  479764 ns/op | 273.20 MB/s | 280209 B/op	|   93 allocs/op |
+| ProtoHTTP1_1K-4  |  30000 |  332801 ns/op |   6.15 MB/s |  52926 B/op	|  399 allocs/op |
+| ProtoHTTP1_64K-4 |   5000 | 1419998 ns/op |  92.30 MB/s | 929968 B/op	| 1080 allocs/op |
+| ProtoHTTP2_1K-4  |  50000 |  128962 ns/op |  15.88 MB/s |  89091 B/op	|  107 allocs/op |
```
Comment (petermattis) [inline README.md]: Strange that GRPC_1K-4 is faster?

**Comment by petermattis** [comment]
LGTM

File: README.md (hunk 1 above)
Comment (tamird) [inline README.md]: I believe the implementation isn't shared.

Comment (tschottdorf) [inline README.md]: the http2 implementation isn't shared?

Comment (bdarnell) [inline README.md]: Parts of it are: https://github.com/grpc/grpc-go/blob/174192fc93efcb188fc8f46ca447f0da606b6885/transport/http2_server.go#L46

But grpc doesn't use `net/http`; it builds its own implementation off the lower-level interfaces (line 211 in the above link). So apparently `net/http` has some higher per-request overhead, while grpc has more per-byte overhead in larger requests.

**Pull request merged**